Orthtree/include/CGAL/Orthtree.h

```diff
@@ -43,258 +48,224 @@
-  typedef typename Traits::FT FT; ///< Number type.
-  typedef typename Traits::Point_d Point; ///< Point type.
-  typedef typename Traits::Bbox_d Bbox; ///< Bounding box type.
-  typedef typename Traits::Sphere_d Sphere; ///< Sphere type.
+  using Dimension = typename Traits::Dimension; ///< Dimension of the tree
+  using Kernel = typename Traits::Kernel; ///< Kernel type.
+  using FT = typename Traits::FT; ///< Number type.
+  using Point = typename Traits::Point_d; ///< Point type.
+  using Bbox = typename Traits::Bbox_d; ///< Bounding box type.
+  using Sphere = typename Traits::Sphere_d; ///< Sphere type.
+  using Adjacency = typename Traits::Adjacency; ///< Adjacency type.
+
+  using Node_data = typename Traits::Node_data;
 
-  /// \cond SKIP_IN_MANUAL
-  typedef typename Traits::Array Array; ///< Array type.
-  typedef typename Traits::Construct_point_d_from_array
-  Construct_point_d_from_array;
-  typedef typename Traits::Construct_bbox_d
-  Construct_bbox_d;
-  /// \endcond
   /// @}
 
   /// \name Public Types
   /// @{
 
   /*!
-   * \brief Self typedef for convenience.
+   * \brief Self alias for convenience.
    */
-  typedef Orthtree<Traits, PointRange, PointMap> Self;
+  using Self = Orthtree<Traits>;
 
   /*!
    * \brief Degree of the tree (number of children of non-leaf nodes).
    */
-  typedef Dimension_tag<(2 << (Dimension::value-1))> Degree;
+  using Degree = Dimension_tag<(2 << (Dimension::value - 1))>;
+
+  /*!
+   * \brief Index of a given node in the tree; the root always has index 0.
+   */
+  using Node_index = std::size_t;
 
   /*!
-   * \brief The Sub-tree / Orthant type.
+   * \brief Optional index of a node in the tree.
+   */
+  using Maybe_node_index = std::optional<Node_index>;
+
+  /*!
+    \brief Set of bits representing this node's relationship to its parent.
+
+    Equivalent to an array of Booleans, where index[0] is whether `x`
+    is greater, index[1] is whether `y` is greater, index[2] is whether
+    `z` is greater, and so on for higher dimensions if needed.
+    Used to represent a node's relationship to the center of its parent.
    */
-  class Node;
+  using Local_coordinates = std::bitset<Dimension::value>;
+
+  /*!
+    \brief Coordinates representing this node's relationship
+    with the rest of the tree.
+
+    Each value `(x, y, z, ...)` of global coordinates is calculated by doubling
+    the parent's global coordinates and adding the local coordinates.
+   */
+  using Global_coordinates = std::array<std::uint32_t, Dimension::value>;
 
   /*!
    * \brief A predicate that determines whether a node must be split when refining a tree.
    */
-  typedef std::function<bool(Node)> Split_predicate;
+  using Split_predicate = std::function<bool(Node_index, const Self&)>;
 
   /*!
-   * \brief A model of `ConstRange` whose value type is `Node`.
+   * \brief A model of `ConstRange` whose value type is `Node_index` and its iterator type is `ForwardIterator`.
```

Comment:
Is it really only `ForwardIterator` ?

Reply:
Yes, based on a lambda "next" function. See Orthtree::traverse() for details.

Reply:
What about  https://doc.cgal.org/latest/Manual/classForwardRange.html

Reply:
done